x/incentive/client/cli/tx.go

```diff
@@ -22,292 +18,151 @@ const multiplierFlag = "multiplier"
+			fmt.Sprintf(`  $ %s tx %s claim-hard --%s hard=large,ukava=small`, version.AppName, types.ModuleName, multiplierFlag),
 		}, "\n"),
 		Args: cobra.NoArgs,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			inBuf := bufio.NewReader(cmd.InOrStdin())
-			cliCtx := context.NewCLIContext().WithCodec(cdc)
-			txBldr := auth.NewTxBuilderFromCLI(inBuf).WithTxEncoder(utils.GetTxEncoder(cdc))
-
-			sender := cliCtx.GetFromAddress()
-			selections := types.NewSelectionsFromMap(denomsToClaim)
-
-			msg := types.NewMsgClaimHardReward(sender, selections...)
-			if err := msg.ValidateBasic(); err != nil {
+			cliCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
 				return err
 			}
-			return utils.GenerateOrBroadcastMsgs(cliCtx, txBldr, []sdk.Msg{msg})
-		},
-	}
-	cmd.Flags().StringToStringVarP(&denomsToClaim, multiplierFlag, multiplierFlagShort, nil, "specify the denoms to claim, each with a multiplier lockup")
-	cmd.MarkFlagRequired(multiplierFlag)
-	return cmd
-}
-
-func getCmdClaimHardVVesting(cdc *codec.Codec) *cobra.Command {
-	var denomsToClaim map[string]string
-
-	cmd := &cobra.Command{
-		Use:   "claim-hard-vesting [receiver]",
-		Short: "claim Hard module rewards on behalf of a validator vesting account using given multipliers",
-		Long: `Claim sender's outstanding hard supply/borrow rewards on behalf of a validator vesting account using given multipliers
-A receiver address for the rewards is needed as validator vesting accounts cannot receive locked tokens.`,
-		Example: strings.Join([]string{
-			fmt.Sprintf("  $ %s tx %s claim-hard-vesting kava15qdefkmwswysgg4qxgqpqr35k3m49pkx2jdfnw --%s hard=large --%s ukava=small", version.ClientName, types.ModuleName, multiplierFlag, multiplierFlag),
-			fmt.Sprintf("  $ %s tx %s claim-hard-vesting kava15qdefkmwswysgg4qxgqpqr35k3m49pkx2jdfnw --%s hard=large,ukava=small", version.ClientName, types.ModuleName, multiplierFlag),
-		}, "\n"),
-		Args: cobra.ExactArgs(1),
-		RunE: func(cmd *cobra.Command, args []string) error {
-			inBuf := bufio.NewReader(cmd.InOrStdin())
-			cliCtx := context.NewCLIContext().WithCodec(cdc)
-			txBldr := auth.NewTxBuilderFromCLI(inBuf).WithTxEncoder(utils.GetTxEncoder(cdc))
 
 			sender := cliCtx.GetFromAddress()
-			receiver, err := sdk.AccAddressFromBech32(args[1])
-			if err != nil {
-				return err
-			}
 			selections := types.NewSelectionsFromMap(denomsToClaim)
 
-			msg := types.NewMsgClaimHardRewardVVesting(sender, receiver, selections...)
+			msg := types.NewMsgClaimHardReward(sender.String(), selections)
 			if err := msg.ValidateBasic(); err != nil {
 				return err
 			}
-			return utils.GenerateOrBroadcastMsgs(cliCtx, txBldr, []sdk.Msg{msg})
+			return tx.GenerateOrBroadcastTxCLI(cliCtx, cmd.Flags(), &msg)
 		},
 	}
 	cmd.Flags().StringToStringVarP(&denomsToClaim, multiplierFlag, multiplierFlagShort, nil, "specify the denoms to claim, each with a multiplier lockup")
 	cmd.MarkFlagRequired(multiplierFlag)
```

Comment:
`cmd.MarkFlagRequired()` returns an error, seems like cosmos just discards the error without handling it, but maybe we can just panic for it

Reply:
good idea, i'll update to panic here